app/websocket_router.go

```diff
@@ -54,17 +54,17 @@ func (wr *WebSocketRouter) ServeWebSocket(conn *WebConn, r *model.WebSocketReque
 			return
 		}
 
-		wr.app.Srv().Go(func() {
-			wr.app.SetStatusOnline(session.UserId, false)
-			wr.app.UpdateLastActivityAtIfNeeded(*session)
-		})
-
 		conn.SetSession(session)
 		conn.SetSessionToken(session.Token)
 		conn.UserId = session.UserId
 
 		wr.app.HubRegister(conn)
 
+		wr.app.Srv().Go(func() {
+			wr.app.SetStatusOnline(session.UserId, false)
+			wr.app.UpdateLastActivityAtIfNeeded(*session)
+		})
```

Comment:
If I understood the problem correctly, there's a race between sending the status change message and registering the connection in the hub. If that's the case I am not sure how changing the order here can fully correct the issue.

As far as I can see we are still lacking any guarantee that the connection is going to get registered before we try to send the status change message.

Reply:
And that's why I set the "Do not merge" label. :wink:

Reply:
Yes, just though it was ready for review. I am guessing https://github.com/mattermost/mattermost-server/pull/14399 is the other piece of the puzzle.

Reply:
Yes, it _is_ ready for review. Considering #14399 is applied. Sorry I should have made that clear :P

Reply:
No worries :+1: , just making sure we are not missing anything. I could have guessed but @isacikgoz was totally unaware of that one and I fear he might have not been presented with the full picture.